### ISSUE TITLE
Is that possible to generate unique login which is similar to the original data ?

### ISSUE THREAD

**Issue by zengheng3** [DivanteLtd/anonymizer#41]
Hello,

I just tried this tool and it worked well on names and emails.
But for the field ‘login’, I have to use the type 'uniq_login' which generates something like UUID.
If I use the 'login' type, the generated data is similar to my original data, but SQL ('Duplicate Entry') errors appear and interrupt the execution.
In my mind, any field with the name "login" must be unique, and this is also part of its definition.
So, I would like to ask if it is possible to create a unique login similar to the original one.

**Comment by mkoszutowski** [comment]
Hi, what do you mean "create a unique login similar to the original one"? Now, uniq_login is an md5 hash. Why? Because is almost unique, is alphanumeric, is easy to create. The last argument is the most important. If you have an easy way to generate unique, alphanumeric, similar to the original one by mysql method I would implement it. Unfortunately, now I don't have an idea better than an md5 hash.

**Comment by zengheng3** [comment]
You are right, for now I haven't yet found a better solution, so I turned to you. I saw this function in some other libraries but they just store the generated data every time you call the function 'unique()', and this is not fast enough for big tables.
